<?xml version='1.0' encoding='UTF-8'?>
<root version="2.3.8">
 <tabbed_widget name="Main Window" parent="main_window">
  <plotmatrix columns="1" rows="2" tab_name="robot1">
   <plot col="0" style="Lines" row="0">
    <range bottom="4.469836" top="5.095037" right="-53.690000" left="-58.700000"/>
    <limitY/>
    <curve name="/robot1/got_pose/pose/pose/position/x" R="23" G="190" custom_transform="noTransform" B="207"/>
    <curve name="/robot1/got_pose_corrected/pose/pose/position/x" R="31" G="119" custom_transform="noTransform" B="180"/>
    <curve name="/robot1/ground_truth/pose/position/x" R="214" G="39" custom_transform="noTransform" B="40"/>
    <curve name="/robot1/odom_global_got_filtered/pose/position/x" R="26" G="201" custom_transform="noTransform" B="56"/>
    <curve name="/robot1/odom_global_got_no_thresh_filtered/pose/position/x" R="188" G="189" custom_transform="noTransform" B="34"/>
    <curve name="/robot1/odom_local_filtered/pose/position/x" R="148" G="103" custom_transform="noTransform" B="189"/>
    <transform value="noTransform"/>
   </plot>
   <plot col="0" style="Lines" row="1">
    <range bottom="3.169449" top="3.495660" right="-53.690000" left="-58.700000"/>
    <limitY/>
    <curve name="/robot1/got_pose/pose/pose/position/y" R="23" G="190" custom_transform="noTransform" B="207"/>
    <curve name="/robot1/got_pose_corrected/pose/pose/position/y" R="31" G="119" custom_transform="noTransform" B="180"/>
    <curve name="/robot1/ground_truth/pose/position/y" R="214" G="39" custom_transform="noTransform" B="40"/>
    <curve name="/robot1/odom_global_got_filtered/pose/position/y" R="26" G="201" custom_transform="noTransform" B="56"/>
    <curve name="/robot1/odom_global_got_no_thresh_filtered/pose/position/y" R="188" G="189" custom_transform="noTransform" B="34"/>
    <curve name="/robot1/odom_local_filtered/pose/position/y" R="148" G="103" custom_transform="noTransform" B="189"/>
    <transform value="noTransform"/>
   </plot>
  </plotmatrix>
  <plotmatrix columns="1" rows="2" tab_name="robot2">
   <plot col="0" style="Lines" row="0">
    <range bottom="1.358844" top="2.241843" right="39.000000" left="34.000000"/>
    <limitY/>
    <curve name="/robot2/got_pose_corrected_client/pose/pose/position/x" R="31" G="119" custom_transform="noTransform" B="180"/>
    <curve name="/robot2/ground_truth/pose/position/x" R="214" G="39" custom_transform="noTransform" B="40"/>
    <transform value="noTransform"/>
   </plot>
   <plot col="0" style="Lines" row="1">
    <range bottom="-1.124412" top="-1.059144" right="39.000000" left="34.000000"/>
    <limitY/>
    <curve name="/robot2/got_pose_corrected_client/pose/pose/position/y" R="31" G="119" custom_transform="noTransform" B="180"/>
    <curve name="/robot2/ground_truth/pose/position/y" R="214" G="39" custom_transform="noTransform" B="40"/>
    <transform value="noTransform"/>
   </plot>
  </plotmatrix>
  <currentPlotMatrix index="1"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default time_axis=""/>
  </plugin>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <use_renaming_rules value="true"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <use_renaming_rules value="true"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
  </plugin>
  <plugin ID="ROS /rosout Visualization" status="idle"/>
  <plugin ID="ROS Topic Re-Publisher" status="idle"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets>
  <snippet language="JS" name="1st_derivative">
   <global>var prevX = 0
var prevY = 0</global>
   <equation>dx = time - prevX
dy = value - prevY
prevX = time
prevY = value

return dy/dx</equation>
  </snippet>
  <snippet language="JS" name="1st_order_lowpass">
   <global>var prevY = 0
var alpha = 0.1</global>
   <equation>prevY = alpha * value + (1.-alpha) * prevY

return prevY</equation>
  </snippet>
  <snippet language="JS" name="2D_velocity">
   <global>var prev_x = 0
var prev_y = 0
var prev_t = 0</global>
   <equation>X = $$your_odometry/position/x$$
Y = $$your_odometry/position/y$$

var dist = sqrt( (X-prev_x)*(X-prev_x) + (Y-prev_y)*(Y-prev_y) )
var dT = time - prev_t

prev_x = X
prev_y = Y
prev_t = time

return dist / dT</equation>
  </snippet>
  <snippet language="JS" name="average_two_curves">
   <global></global>
   <equation>a = $$PLOT_A$$
b = $$PLOT_B$$

return (a+b)/2</equation>
  </snippet>
  <snippet language="JS" name="integral">
   <global>var integral = 0</global>
   <equation>integral += value
return integral</equation>
  </snippet>
  <snippet language="JS" name="rad_to_deg">
   <global></global>
   <equation>return value*180/3.1417</equation>
  </snippet>
  <snippet language="JS" name="remove_offset">
   <global>var is_first = true
var first_value = 0</global>
   <equation>if (is_first)
{
  is_first = false
  first_value = value
}

return value - first_value</equation>
  </snippet>
  <snippet language="JS" name="yaw_from_quaternion">
   <global>// source: https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles

function quaternionToYaw(x, y, z, w)
{
  // yaw (z-axis rotation)
  t1 = 2.0 * (w * z + x * y);
  t2 = 1.0 - 2.0 * (y * y + z * z);
  yaw = Math.atan2(t1, t2);

  return yaw
}</global>
   <equation>return quaternionToYaw(x, y, z, w);</equation>
  </snippet>
 </snippets>
 <!-- - - - - - - - - - - - - - - -->
</root>

